test/dtest/ccmlib/scylla_node.py

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from test.pylib.internal_types import ServerInfo
+    from test.dtest.ccmlib.scylla_cluster import ScyllaCluster
+
+
+class ScyllaNode:
```

Comment:
I don't think we want to build yet another test framework inside the test framework.

Things like `grep_log` etc., the test.py topology framework already supports

We should adapt the ported tests to the existing framework... not the other way around

Reply:
There are 150,000 lines of code in dtest. If you want to move the dtest tests into the Scylla repository without ccm (I didn't vote for doing this, but it seems it was decided), I don't see any other viable approach accept implementing dtest-like and ccm-like features using a wrapper using whatever pylib features you prefer. Converting all these function calls manually to the slightly-different version used in pylib will take agest.